.github/workflows/build.yaml

```diff
@@ -55,12 +55,46 @@ jobs:
       ocm-labels: ${{ toJSON(matrix.args.ocm-labels) }}
       extra-tags: ${{ inputs.mode == 'release' && 'latest' || '' }}
 
-  verify:
+  verify-others:
+      name: Verify (${{ matrix.task }})
+      runs-on: ubuntu-latest
+      permissions:
+        contents: read
+      env:
+        GO_TEST_RACE: 1
+        GOLANGCI_LINT_VERBOSE: 1
+        GOLANGCI_LINT_TIMEOUT: 5m
+      strategy:
+        fail-fast: false
+        matrix:
+          task: ["check-generate", "lint", "go-test"]
+      steps:
+        - name: Trusted Checkout
+          uses: gardener/cc-utils/.github/actions/trusted-checkout@master
+
+        - name: Set up Go
+          uses: actions/setup-go@v5
+          with:
+            go-version-file: 'go.mod'
+            cache: true
+
+        - name: Set Git user email
+          if: matrix.task == 'check-generate'
+          run: git config --global user.email "gardener@sap.com"
```

Comment:
you might use https://github.com/gardener/cc-utils/tree/master/.github/actions/setup-git-identity